public/app/features/dashboard-scene/serialization/transformSceneToSaveModelSchemaV2.ts

```diff
@@ -370,6 +380,36 @@ function getVariables(oldDash: DashboardSceneState) {
   return variables;
 }
 
+function getAnnotations(state: DashboardSceneState): AnnotationQueryKind[] {
+  const data = state.$data;
+  if (!(data instanceof DashboardDataLayerSet)) {
+    return [];
+  }
+  const annotations: AnnotationQueryKind[] = [];
+  for (const layer of data.state.annotationLayers) {
+    if (!(layer instanceof dataLayers.AnnotationsDataLayer)) {
+      continue;
+    }
+    const result: AnnotationQueryKind = {
+      kind: 'AnnotationQuery',
+      spec: {
+        name: layer.state.query.name,
+        datasource: layer.state.query.datasource!, // FIXME: this cannot be undefined
+        query: {
+          kind: layer.state.query.datasource?.type ?? 'default',
```

Comment:
So this needs cannot be 'default', it needs to be a plugin name. With the lack of the datasource provided we need to resolve the type of the default datasource.

Reply:
If we assume we want to handle `undefined` as default, yes. If we want to have a keyword, this `default` represents that keyword.

Reply:
do you mean we need to resolve the data source during the transformation?

Reply:
I think so - then you can retrieve the type of the data source.

Reply:
This makes the whole process async, so it is not possible. I think we need to leave an empty object as a datasource type, and it is resolved in runtime once the scene is mounted. If DS service receives undefined or non-valid DS it falls back to default

Reply:
Hm, but this means that we produce invalid kind and end up with incorrect schema. I guess the async thing makes it hard because we need to update many places to suddenly become async too?

Reply:
yes. This translation should be synchronous, otherwise will make any process translating async, like creating a scene. The real thing is, that we have many async resolving for Scene, but once the scene is built and activated. So we should leave the scene to figure out what to put there when the kind is not defined. 

This is, in the end defining the type of the query. It is weird to not know it in advance, because even if the default is X, the type should be known in advance. So, falling back to default doesn't make much sense.

I don't really know how to solve this use case, tbh

Reply:
@ivanortegaalba - first thing that comes to my mind is that we resolve the default datasource and keep in the dashboard scene state for retireval later on when serialising the state. Maybe we could keep it as part of the serialiser I'm proposing in https://github.com/grafana/grafana/pull/96931.

Alternatively we could allow providing a context to this function call, basically a 2nd argument, and then we would not have to make it async at all. However, whoever calls this function would have to be async.

Reply:
>  It is weird to not know it in advance, because even if the default is X, the type should be known in advance

Well, technically, the default datasource is known at that time because it needs to be resolved for the queries to be run... It's just the function definition that makes it async...

But - the default data source can be resolved from boot config. There's the `defaultDatasource` uid in the config and the list of all the datasources from which we can figure out the default one synchroniously.